flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/maintenance/operator/TableChange.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.flink.maintenance.operator;
+
+import java.util.Objects;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.DeleteFile;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+
+/** Event describing changes in an Iceberg table */
+class TableChange {
+  private int dataFileNum;
+  private int deleteFileNum;
+  private long dataFileSize;
+  private long deleteFileSize;
+  private int commitNum;
+
+  TableChange(
+      int dataFileNum, int deleteFileNum, long dataFileSize, long deleteFileSize, int commitNum) {
+    this.dataFileNum = dataFileNum;
+    this.deleteFileNum = deleteFileNum;
+    this.dataFileSize = dataFileSize;
+    this.deleteFileSize = deleteFileSize;
+    this.commitNum = commitNum;
+  }
+
+  TableChange(Snapshot snapshot, FileIO io) {
+    Iterable<DataFile> dataFiles = snapshot.addedDataFiles(io);
+    Iterable<DeleteFile> deleteFiles = snapshot.addedDeleteFiles(io);
+
+    dataFiles.forEach(
+        dataFile -> {
+          this.dataFileNum++;
+          this.dataFileSize += dataFile.fileSizeInBytes();
+        });
+
+    deleteFiles.forEach(
+        deleteFile -> {
+          this.deleteFileNum++;
+          this.deleteFileSize += deleteFile.fileSizeInBytes();
+        });
+
+    this.commitNum = 1;
+  }
+
+  int dataFileNum() {
```

Comment:
is this used?

Reply:
Will be used later 😄 
Since `TableChange` is a POJO like object, I have defined all the accessor methods